priv/tools/internal/block_audit.erl

```diff
@@ -0,0 +1,236 @@
+    Buckets = case proplists:get_all_values(bucket, Opts) of
+                  [] ->
+                      {ok, AllBuckets} = riakc_pb_socket:list_keys(Pid, ?BUCKETS_BUCKET),
+                      AllBuckets;
+                  Values ->
+                      Values
+              end,
+    info("Retrieved bucket list. There are ~p buckets, including tombstones.",
+         [length(Buckets)]),
+    info("Searching for orphaned blocks. This may take a while...", []),
+    log_all_maybe_orphaned_blocks(Pid, Opts, Buckets),
+    ok.
+
+log_all_maybe_orphaned_blocks(_Pid, _Opts, []) ->
+    ok;
+log_all_maybe_orphaned_blocks(Pid, Opts, [Bucket | Buckets]) ->
+    _ = log_maybe_orphaned_blocks(Pid, Opts, Bucket),
+    log_all_maybe_orphaned_blocks(Pid, Opts, Buckets).
+
+log_maybe_orphaned_blocks(Pid, Opts, Bucket) when is_binary(Bucket) ->
+    log_maybe_orphaned_blocks(Pid, Opts, binary_to_list(Bucket));
+log_maybe_orphaned_blocks(Pid, Opts, Bucket) ->
+    info("Finding Orphaned blocks for Bucket ~p", [Bucket]),
+    BlocksTable = list_to_atom(Bucket),
+    ets:new(BlocksTable, [set, named_table, public, {keypos, #buuid.uuid}]),
+    try
+        {ok, NumKeys} = cache_block_keys(Pid, Bucket, BlocksTable),
+        case NumKeys of
+            0 -> ok;
+            _ -> case delete_manifest_uuids(Pid, Bucket, BlocksTable) of
+                     ok ->
+                         write_uuids(Opts, Bucket,
+                                     BlocksTable, ets:info(BlocksTable, size));
+                     _ ->
+                         nop
+                 end
+        end
+    after
+        catch ets:delete(BlocksTable)
+    end.
+
+write_uuids(_Opts, _Bucket, _BlocksTable, 0) ->
+    ok;
+write_uuids(Opts, Bucket, BlocksTable, _) ->
+    OutDir = proplists:get_value(output, Opts),
+    Filename = filename:join(OutDir, Bucket),
+    ok = filelib:ensure_dir(Filename),
+    {ok, File} = file:open(Filename, [write, raw, delayed_write]),
+    {UUIDs, Blocks} = ets:foldl(
+                        fun(#buuid{uuid=UUID, seqs=Seqs},
+                            {TotalUUIDs, TotalBlocks}) ->
+                                verbose(Opts, "~s ~s ~B ~p",
+                                        [Bucket, mochihex:to_hex(UUID),
+                                         length(Seqs), Seqs]),
+                                [ok = file:write(File,
+                                                 [Bucket, $ ,
+                                                  mochihex:to_hex(UUID), $ ,
+                                                  integer_to_list(Seq), $\n]) ||
+                                    Seq <- Seqs],
+                                {TotalUUIDs + 1, TotalBlocks + length(Seqs)}
+                        end, {0, 0}, BlocksTable),
+    ok = file:close(File),
+    info("Total # Missing Block UUIDs: ~p [count]", [UUIDs]),
+    info("Total # Orphaned Blocks: ~p [count]", [Blocks]),
+    info("Orphaned Blocks written to ~p", [Filename]).
```

Comment:
I would like to see this as absolute path as `riak-cs script` command changed current directory and it's output there.